util/movement-types/src/transaction.rs

```diff
@@ -40,13 +41,15 @@ impl fmt::Display for Id {
 	}
 }
 
-#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash)]
+#[derive(
+	Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, CryptoHasher, BCSCryptoHash,
+)]
 pub struct Transaction {
 	data: Vec<u8>,
 	// Application priority is stored low to high, i.e., 0 is the highest priority.
 	application_priority: u64,
 	sequence_number: u64,
-	id: Id,
+	pub id: Id,
```

Comment:
The pub member now sticks out like a sore thumb. At least in movement-types I'd push for better API style.